File: Creational Patterns/java/factoryMethod/src/main/java/factorymethod/MySQLDatabase.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package factorymethod;

/**
 *
 * @author julianm
 */
public class MySQLDatabase implements IDatabaseConnection {

    @Override
    public void connect() {
        System.out.println("Conectando a la base de datos MySQL...");
    }

    @Override
    public void disconnect() {
        System.out.println("Desconectando de la base de datos MySQL...");
    }
    
}

File: Creational Patterns/typeScript/src/class/MySQLDatabase.ts
import { IDatabaseConnection } from "../interface/IDatabaseConnection";

// Clase que implementa la interfaz IDatabaseConnection
export class MySQLDatabase implements IDatabaseConnection {
    connect(): void {
        console.log("Conectando a la base de datos MySQL...");
    }

    disconnect(): void {
        console.log("Desconectando de la base de datos MySQL...");
    }
}